circuitpython_build_tools/scripts/build_bundles.py

```diff
@@ -49,47 +49,54 @@ def add_file(bundle, src_file, zip_name):
 
 
 def build_bundle(libs, bundle_version, output_filename,
-        build_tools_version="devel", mpy_cross=None):
+        build_tools_version="devel", mpy_cross=None, example_bundle=False):
     build_dir = "build-" + os.path.basename(output_filename)
-    build_lib_dir = os.path.join(build_dir, "lib")
+    build_lib_dir = os.path.join(build_dir, build_dir.replace(".zip", ""), "lib")
+    build_example_dir = os.path.join(build_dir, build_dir.replace(".zip", ""), "examples")
     if os.path.isdir(build_dir):
         print("Deleting existing build.")
         shutil.rmtree(build_dir)
-    os.makedirs(build_lib_dir)
+    total_size = 0
+    if not example_bundle:
+        os.makedirs(build_lib_dir)
+        total_size += 512
+    os.makedirs(build_example_dir)
+    total_size += 512
 
     multiple_libs = len(libs) > 1
 
     success = True
-    total_size = 512
     for library_path in libs:
         try:
-            build.library(library_path, build_lib_dir, mpy_cross=mpy_cross)
+            build.library(library_path, build_lib_dir, mpy_cross=mpy_cross,
+                          example_bundle=example_bundle)
         except ValueError as e:
-            print(library_path)
+            print("build.library failure:", library_path)
             print(e)
             success = False
 
     print()
-    print("Generating VERSIONS")
-    if multiple_libs:
-        with open(os.path.join(build_lib_dir, "VERSIONS.txt"), "w") as f:
-            f.write(bundle_version + "\r\n")
-            versions = subprocess.run('git submodule foreach \"git remote get-url origin && git describe --tags\"', shell=True, stdout=subprocess.PIPE)
-            if versions.returncode != 0:
-                print("Failed to generate versions file. Its likely a library hasn't been "
-                      "released yet.")
-                success = False
-            
-            repo = None
-            for line in versions.stdout.split(b"\n"):
-                if line.startswith(b"Entering") or not line:
-                    continue
-                if line.startswith(b"git@"):
-                    repo = b"https://github.com/" + line.split(b":")[1][:-len(".git")]
-                elif line.startswith(b"https:"):
-                    repo = line.strip()[:-len(".git")]
-                else:
-                    f.write(repo.decode("utf-8", "strict") + "/releases/tag/" + line.strip().decode("utf-8", "strict") + "\r\n")
+    if not example_bundle:
+        print("Generating VERSIONS")
+        if multiple_libs:
```

Comment:
Don't we want the versions file in the example bundle as well?

Reply:
Two things led me to not include it:
1. It was erroring early on when I setup `example_bundle`, and separated `build_lib_dir`/`build_example_dir` so that the `lib` folder wasn't created in the example bundle.
2. After getting the above working, I glanced at it and thought it wouldn't add anything to examples. That was a personal judgement call, and I'll be the first to call it the wrong one if that's the case. :smile:

It _should_ be an easy fix. Just need to switch its call for `build_lib_dir` to `build_dir`, and expand the `if multiple libs` condition. I'll need to re-create my local code...because I was not smart and didn't save it. 🤦‍♂️

Reply:
So, required a little more local code glue-in and cloning the actual CP bundle to get a verifiable result. I currently have it placing `VERSIONS` at the same level as the `example` and `libs` folders:

```shell
sommersoft@thespacebetween:~/Dev/cpy-build-tools/circuitpython-build-tools$ unzip -l ./bundles/local-test-examples-d4044dc.zip
Archive:  ./bundles/local-test-examples-d4044dc.zip
{"build-tools-version": "devel"}
  Length      Date    Time    Name
---------  ---------- -----   ----
     6014  2018-10-13 10:20   build_local_test_examples_d4044dc/VERSIONS.txt
     1062  2018-10-13 10:20   build_local_test_examples_d4044dc/examples/struct.py
     1133  2018-10-13 10:20   build_local_test_examples_d4044dc/examples/hello_CircuitPython_I2C.py
      397  2018-10-13 10:20   build_local_test_examples_d4044dc/examples/max31855_simpletest.py
     2885  2018-10-13 10:20   build_local_test_examples_d4044dc/examples/snapshot_internal.py
      426  2018-10-13 10:20   build_local_test_examples_d4044dc/examples/ccs811_simpletest.py
```
```shell
sommersoft@thespacebetween:~/Dev/cpy-build-tools/circuitpython-build-tools$ unzip -l ./bundles/local-test-py-d4044dc.zip
Archive:  ./bundles/local-test-py-d4044dc.zip
{"build-tools-version": "devel"}
  Length      Date    Time    Name
---------  ---------- -----   ----
     6014  2018-10-13 10:20   build_local_test_py_d4044dc/VERSIONS.txt
    17681  2018-10-13 10:20   build_local_test_py_d4044dc/lib/adafruit_ina219.py
.......
     1062  2018-10-13 10:20   build_local_test_py_d4044dc/examples/struct.py
```

Reply:
Looks good!